src/compile/mark/mark.ts

```diff
@@ -60,12 +60,12 @@ function parsePathMark(model: UnitModel) {
   const details = detailFields(model);
 
   const role = model.markDef.role || markCompiler[mark].defaultRole;
-
+  const clip = model.markDef.clip != null ? model.markDef.clip : scaleClip(model);
```

Comment:
I only mention `model.markDef.clip !== undefined` because I thought you intend to use `scaleClip()` function only if clip is not specified (`undefined`).   (If users directly specify `clip`, then we should follow their instruction.)  

For `null` case, I think it's fair to interpret as `false`.  I also believe that passing `null` to Vega would be equivalent to false since underlying Vega probably would clip when `clip` is truthy?   If so, we might not need any special casing for `null`.